Lib/multiprocessing/pool.py

```diff
@@ -429,6 +450,17 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         )
         return result
 
+    def _wait_for_updates(self, timeout=None):
+        task_queue_sentinels = [self._outqueue._reader]
+        self_notifier_sentinels = [self._change_notifier._reader]
+        worker_sentinels = [worker.sentinel for worker in self._pool]
+        sentinels = [*task_queue_sentinels,
+                     *worker_sentinels,
+                     *self_notifier_sentinels]
+        wait(sentinels, timeout=timeout)
```

Comment:
Can you please add a comment on wait() to explain that it completes when at least one sentinel is set and that it's important to not wait until all sentinels completed, but exit frequently to refresh the pool.

This point is non-trivial and it surprised me when I wrote PR #11136, my comment https://github.com/python/cpython/pull/11136#issuecomment-447291966:
> My change doesn't work: self._worker_state_event isn't set when a worker completes, whereas _maintain_pool() should be called frequently to check when a worker completed.

Reply:
`wait` is already documented: https://docs.python.org/3/library/multiprocessing.html#multiprocessing.connection.wait

Reply:
I don't ask to documen the behavior of wait, but more explicit that we stop as soon as the first event complete on purpose.

Reply:
It looks obvious to me, especially as the function is named `wait_for_updates`, but I guess it doesn't hurt to add a comment.